docs/contracts/v2/guides/proxy-architecture/04-batch-stream.mdx

```diff
@@ -0,0 +1,297 @@
+inefficient due to multiple ERC-20 transfers from the Proxy owner to the Proxy, so we pass the token as a parameter in
+the function, without encapsulating it in each struct of the array.
+
+## Batch create functions
+
+There are four batch create functions in the Proxy Target contract, two for each Lockup contract:
+
+Lockup Linear:
+
+- [`batchCreateWithDurations`](/contracts/v2/reference/periphery/contract.SablierV2ProxyTarget#batchcreatewithdurations)
+- [`batchCreateWithRange`](/contracts/v2/reference/periphery/contract.SablierV2ProxyTarget#batchcreatewithrange)
+
+Lockup Dynamic:
+
+- [`batchCreateWithDeltas`](/contracts/v2/reference/periphery/contract.SablierV2ProxyTarget#batchcreatewithdeltas)
+- [`batchCreateWithMilestones`](/contracts/v2/reference/periphery/contract.SablierV2ProxyTarget#batchcreatewithmilestones)
+
+Let's start with a Lockup Linear example:
+
+## Set up a contract
+
+Declare the Solidity version used to compile the contract:
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.19;
+```
+
+As mentioned in the [Overview](/contracts/v2/guides/proxy-architecture/overview), we use `Permit2` for all token
+interactions. In our example we will deal with a contract creator, and contracts can't sign transactions. Having that
+said, we will need an implementation of the [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) standard.
+
+Let's declare a dummy contract `ERC1271` which returns the magic value (the function signature of `isValidSignature`):
+
+```solidity
+/// @notice Permit2 uses this ERC to validate contract signatures.
+contract ERC1271 {
+    function isValidSignature(bytes32, /* hash */ bytes memory /* signature */ ) public pure returns (bytes4) {
+        return this.isValidSignature.selector;
+    }
+}
+```
+
+Import the relevant symbols from `@sablier/v2-core` and `@sablier/v2-periphery`:
+
+```solidity
+import { ISablierV2LockupLinear } from "@sablier/v2-core/interfaces/ISablierV2LockupLinear.sol";
+import { LockupLinear } from "@sablier/v2-core/types/DataTypes.sol";
+import { ud60x18 } from "@sablier/v2-core/types/Math.sol";
+import { IERC20 } from "@sablier/v2-core/types/Tokens.sol";
+import { ISablierV2ProxyTarget } from "@sablier/v2-periphery/interfaces/ISablierV2ProxyTarget.sol";
+import { Batch, Broker } from "@sablier/v2-periphery/types/DataTypes.sol";
+import { IAllowanceTransfer, Permit2Params } from "@sablier/v2-periphery/types/Permit2.sol";
+import { IPRBProxy, IPRBProxyRegistry } from "@sablier/v2-periphery/types/Proxy.sol";
+```
+
+Create a contract called `BatchLockupLinearStreamCreator` that inherits `ERC1271`, and declare a constant `DAI` of type
+`IERC20`, a constant `PERMIT2` of type `IAllowanceTransfer`, a constant `PROXY_REGISTRY` of type `IPRBProxyRegistry`, an
+immutable variable `lockupLinear` of type `ISablierV2LockupLinear` and an immutable variable `proxyTarget` of type
+`ISablierV2ProxyTarget`:
```

Comment:
Very verbose, in complex cases like this you can just cut to the chase by saying "and declare the following constants"